### ISSUE TITLE
A <button> needs two clicks when it has hover/focus states (but does not fire another event?)

### ISSUE THREAD

**Issue by tofumatt** [mozilla/addons-frontend#2594]
This is what caused #1904 and is somewhat explained in https://github.com/mozilla/addons-frontend/pull/2591#discussion_r122276326.

Basically a `<button>` element's `onClick` event is not fired until the second click–possibly only when it is not inside a form.

**Comment by kumar303** [comment]
It looks like a [@media pointer](http://caniuse.com/#feat=css-media-interaction) rule that disables hover on mobile [will fix it](https://css-tricks.com/annoying-mobile-double-tap-link-issue/) but this isn't supported in Firefox yet. The [tracking bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1035774) suggests this workaround:

```css
@media (-moz-touch-enabled: 1), (pointer: coarse) {
  // Rules for touch screens
}
```

**Comment by tofumatt** [comment]
Unfortunately what we actually need to do is target non-touch screens with hover styling because it's the `:hover` and `:focus` styles that cause issues for touch displays that don't support `:hover`.

When I tried `@media (-moz-touch-enabled: 0), (pointer: fine) {` it didn't work. I will admit, as a nit, I tried them in the opposite order but that shouldn't matter...

**Comment by kumar303** [comment]
Ah, bummer. We'll have to keep an eye out for this when we add more buttons in the future.

**Comment by muffinresearch** [comment]
@tofumatt can you add a full STR for an example of this issue - it's unclear to me what the impact is. This is only on mobile AFAICT from the linked issue.

**Comment by tofumatt** [comment]
The steps to reproduce are:

1. create a `<button>` element or a `<Button>` component with an `onClick` handler and a `:hover` style
2. On a desktop browser the `onClick` will fire on first click (expected result)
3. On a mobile/touch-based browser the `:hover` state will be activated on first click/tap and the `onClick` event will not fire
4. On the second click/tap the `onClick` event will fire

**Comment by willdurand** [comment]
@kumar303 is it the issue you mentioned in a recent patch?

**Comment by kumar303** [comment]
Yes. My observations in https://github.com/mozilla/addons-frontend/pull/6860 were that this has been fixed in a recent Firefox for Android release (I was testing in FF62). I inferred that the version we originally saw this in was FF58 which is quite old now. 

I asked in `#mobile` if anyone knew what fixed it but no one knew. I hunted for some bugs but couldn't find anything. Both a recent Chrome Android and Safari iOS do not show the hover problem so I think it's safe to say that this is no longer a problem.
